ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/NettyApplicationCallHandler.kt

```diff
@@ -47,7 +50,30 @@ internal class NettyApplicationCallHandler(
         }
     }
 
-    public companion object {
+    companion object {
         private val CallHandlerCoroutineName = CoroutineName("call-handler")
     }
 }
+
+internal fun HttpRequest.isValid(): Boolean {
+    if (decoderResult().isFailure) {
+        return false
+    }
+
+    val encodings = headers().getAll(io.ktor.http.HttpHeaders.TransferEncoding) ?: return true
+
+    encodings.forEachIndexed { index, header ->
+        val chunkedStart = header.indexOf(CHUNKED_VALUE)
+
+        if (chunkedStart == -1) return@forEachIndexed
+
+        if (index + 1 != encodings.size) return false
+
+        val chunkedIsNotLast = chunkedStart + CHUNKED_VALUE.length < header.length
```

Comment:
What if we have `Zchunked`: we are going to fail this but we shouldn't, right? I would also check the beginning index to see 0 index or comma, or something.. that will complicate the function for sure 😔

Reply:
The problem is that a single entry possibly contains multiple encodings, so yep. I will add a couple of unit tests to check this method